### PR TITLE
First iteration of mobile guides

### PR DESCRIPTION
**Pull request by jcleblanc**

### CODE REVIEW

File: content/guides/mobile/android/index.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+related_pages:
+ - sdks-and-tools
+---
+
+# Android
+
+The [Box Android SDK][android-sdk] provides native access to the Box API from
+within your Android project.
+
```
Comment (cbetta) [inline content/guides/mobile/android/index.md]: Definitely needs some more content. Ideas:

* Image to show what you can do?
* Maybe an overview of what the SDK enables (and doesn't enable?)

File: content/guides/mobile/android/install.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+ - sdks-and-tools
+---
+
+# Android SDK Installation
+
+The Android SDK may be obtained through several methods:
+
+* Adding as a Maven or Gradle dependency
+* Cloning the source into your project
+* Downloading one of the precompiled JARs
```
Comment (cbetta) [inline content/guides/mobile/android/install.md]: Does the SDK itself have further instructions? Is it worth linking there? Maybe use a CTA block at the end of the page?

File: content/guides/mobile/index.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+related_pages:
+ - sdks-and-tools
+---
+
+# Mobile
```
Comment (cbetta) [inline content/guides/mobile/index.md]: Definitely needs more content. Ideas:

* Image to show what the SDKs allow a dev to do?
* Maybe an overview of what each SDK enables (and doesn't enable?)

File: content/guides/mobile/ios/index.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+---
+related_pages:
+ - sdks-and-tools
+---
+
+# iOS
```
Comment (cbetta) [inline content/guides/mobile/ios/index.md]: Definitely needs more content. Ideas:

* Image to show what the SDK allows a dev to do?
* Maybe an overview of the SDK enables (and doesn't enable?)

File: content/guides/mobile/ios/quick-start/0-index.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+type: quick-start
+hide_step_number: true
+hide_in_page_nav: true
+---
+
+# iOS SDK Quick Start
+
+The Box iOS SDK is a [Swift language](https://developer.apple.com/swift/)
+wrapper for adding the Box API into your iOS project.
+
```
Comment (cbetta) [inline content/guides/mobile/ios/quick-start/0-index.md]: Maybe add a nice image like we've done with both the Metadata and Postman quick starts?

File: content/guides/mobile/ios/quick-start/1-create-ios-app.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
+---
+type: quick-start
+hide_in_page_nav: true
+---
+
+# Create an iOS App
+
+Before installing the Box iOS SDK you'll need to have an iOS application
+available to add the dependencies into. We'll create a new blank application
+now.
```
Comment (cbetta) [inline content/guides/mobile/ios/quick-start/1-create-ios-app.md]: "We'll create a new blank application now" yet then we give people a choice to either use a blank app or an existing one.

File: content/guides/mobile/ios/quick-start/1-create-ios-app.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+       ![Configure iOS application](./create-app-config.png)
+     </ImageFrame>
+  5. Select a local storage location for the application and click **Create**
+</Choice>
+
+<Choice option='ios.app_setup' value='existing' color='none'>
+  # Use an existing iOS application
+
+  If you have an existing blank application, skip to the next step.
+</Choice>
```
Comment (cbetta) [inline content/guides/mobile/ios/quick-start/1-create-ios-app.md]: Not sure this block is needed.

File: content/guides/mobile/ios/quick-start/1-create-ios-app.md [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+
+<Choice option='ios.app_setup' value='new' color='blue'>
+  # Create a new iOS application
+
+  1. Load Xcode
+  2. From the top menu, select `File` -> `New` -> `Project...`
+  3. Select the option for **Single View App**
+    <ImageFrame border center>
+      ![Select iOS application type](./create-app-type.png)
+    </ImageFrame>
```
Comment (cbetta) [inline content/guides/mobile/ios/quick-start/1-create-ios-app.md]: I think there are some issues here with putting block-elements (imageframe) into a list. 

I would just turn this into an unordered list (*) and indent the `ImageFrame` with plenty of space around it.

```
  * Load Xcode
  * From the top menu, select `File` -> `New` -> `Project...`
  * Select the option for **Single View App**

  <ImageFrame border center>
    ![Select iOS application type](./create-app-type.png)
  </ImageFrame>

  * Enter your application configuration information, including the **Product
    Name**, **Organization Identifier**, and **Organization Name**
```

File: content/guides/mobile/ios/quick-start/1-create-ios-app.md [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+
+    If you have some iOS development experience and already have a blank
+    iOS application created.
+  </Choose>
+</Grid>
+
+<Choice option='ios.app_setup' value='new' color='blue'>
+  # Create a new iOS application
+
+  1. Load Xcode
```
Comment (cbetta) [inline content/guides/mobile/ios/quick-start/1-create-ios-app.md]: Start Xcode?

File: content/guides/mobile/ios/quick-start/1-create-ios-app.md [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+  2. From the top menu, select `File` -> `New` -> `Project...`
```
Comment (cbetta) [inline content/guides/mobile/ios/quick-start/1-create-ios-app.md]: Generally I've been using **Bold** for any text that is a choice and that might need translating.

File: content/guides/mobile/ios/quick-start/2-install-ios-sdk.md [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+    code, aimed at making it easy to share your code and reuse
+    others’ code.
+  </Choose>
+</Grid>
+
+<Choice option='ios.pm_type' value='carthage' color='blue'>
+  # Install the iOS SDK using Carthage
+
+  1. From a terminal window, install Carthage: `brew install carthage`.
+    [See here](https://github.com/Carthage/Carthage#installing-carthage)
```
Comment (cbetta) [inline content/guides/mobile/ios/quick-start/2-install-ios-sdk.md]: Generally "here" is considered bad practice in docs. A better option might be "Head over to the Carthage documentation for other installation methods"

File: content/guides/mobile/ios/quick-start/2-install-ios-sdk.md [hunk 12]
```diff
@@ -0,0 +1,117 @@
...
+    for other installation methods.
+  2. At the root of your iOS application folder, where the `{APPNAME}.xcodeproj`
+     is located, create a new file named `Cartfile`, without an extension.
+  3. Load `Cartfile`, add the **Box iOS SDK** dependency with
```
Comment (cbetta) [inline content/guides/mobile/ios/quick-start/2-install-ios-sdk.md]: Small rewrite: "Open the `Cartfile` and add the **Box iOS SDK** dependency..."

File: content/guides/mobile/ios/quick-start/2-install-ios-sdk.md [hunk 13]
```diff
@@ -0,0 +1,117 @@
...
+     `iOS` directory. This should have a framework file named
+     `BoxSDK.framework`. Keep this window open as we'll use it to add the
+     framework to our project in the next step.
+  6. In your Xcode project, click on the name of your application at the top of
+     the project navigator on the left. In the content that displays, click on
+     the name of your project under the **TARGETS** option. Scroll down to
+     **Frameworks, Libraries, and Embedded Content**. 
+  7. Drag `BoxSDK.framework` from our Finder window over to the frameworks
+     section.
+    <ImageFrame border center>
```
Comment (cbetta) [inline content/guides/mobile/ios/quick-start/2-install-ios-sdk.md]: Similar issues with images here. Again, the block needs to be at the paragraph level (unindented once). As this image is at the end of the list it should then render fine.

File: content/guides/mobile/ios/quick-start/2-install-ios-sdk.md [hunk 14]
```diff
@@ -0,0 +1,117 @@
...
+      ![Add framework to project](./framework-carthage-add.gif)
```
Comment (cbetta) [inline content/guides/mobile/ios/quick-start/2-install-ios-sdk.md]: Not sure why this image isn't loading. My suspicion is it's related to the bad rendering of the block.

File: content/guides/mobile/ios/quick-start/2-install-ios-sdk.md [hunk 15]
```diff
@@ -0,0 +1,117 @@
...
+  1. In your Xcode project, click on the name of your application at the top of
+     the project navigator on the left. In the content that displays, click on
+     the name of your project under the **PROJECT** option.
+  2. Click on **Swift Packages** and click on the `+` to add a package.
+  3. Enter the following repository URL
+     `https://github.com/box/box-ios-sdk.git` and click next.
+  4. Leave the default settings and click next to finish importing.
+  <ImageFrame border center>
+    ![Add framework to project](./import-sdk-spm.gif)
+  </ImageFrame>
```
Comment (cbetta) [inline content/guides/mobile/ios/quick-start/2-install-ios-sdk.md]: Same image issues. Please try adding some whitespace around the `ImageFrame`.

File: content/guides/mobile/ios/quick-start/1-create-ios-app.md [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+    wish to use a new blank application for testing the SDK, this method
+    will guide you through creating a new blank iOS application.
+  </Choose>
+
+  <Choose option='ios.app_setup' value='existing' color='none'>
+    # Use an existing iOS application
+
+    If you have some iOS development experience and already have a blank
+    iOS application created.
+  </Choose>
```
Comment (cbetta) [inline content/guides/mobile/ios/quick-start/1-create-ios-app.md]: Im not sure if we really need to use a `Choose` here. We can just say "You need a project, here is a way to set one up yet if you don't have one already.". It's a bit overkill to have the actual choices.

File: content/guides/mobile/ios/quick-start/3-configure-box-app.md [hunk 17]
```diff
@@ -0,0 +1,82 @@
...
+
+    Create and configure a new Box JWT application from which a developer
+    token may be generated
+  </Choose>
+
+  <Choose option='ios.app_type' value='use_own' color='red'>
+    # Use an existing app
+
+    Use one of your existing Box JWT applications
+  </Choose>
```
Comment (cbetta) [inline content/guides/mobile/ios/quick-start/3-configure-box-app.md]: Im not 100% sure if we really need to use a Choose here. We can just say "You need a Box App, here is a way to set one up yet if you don't have one already.". It's a bit overkill to have the actual choices. 

FYI in the Postman Quick Start we did as this, as it determined if we used our own credentials.

Update: added some comments further down

File: content/guides/mobile/ios/quick-start/3-configure-box-app.md [hunk 18]
```diff
@@ -0,0 +1,82 @@
...
+  1. At the top of the page click the button to **Save Changes**
+</Choice>
+
+<Choice option='ios.app_type' value='use_own' color='blue'>
+  # Use an Existing JWT Box application
+
+  If you have an existing JWT based Box application in your
+  [developer console][devconsole] that you
+  would like to use, skip to the next step.
+</Choice>
```
Comment (cbetta) [inline content/guides/mobile/ios/quick-start/3-configure-box-app.md]: Ok maybe we do need a `Choice` here. Maybe we might want to, expand here to explain how to generate a JWT? They might have created an app already but don't have a JWT keypair yet?

File: content/guides/mobile/ios/quick-start/3-configure-box-app.md [hunk 19]
```diff
@@ -0,0 +1,82 @@
...
+
+Now that an application is available, we need to create a developer token that
+may be used to authenticate the Box iOS SDK to start making calls to the Box
+APIs. 
+
+1. Go to the [Developer Console][devconsole]
+1. Load the application that you would like to use
+1. In the left navigation menu, click on **Configuration**
+1. Under **Developer Token**, click the **Generate Developer Token** button
+1. Copy the token for the next step to make an API call
```
Comment (cbetta) [inline content/guides/mobile/ios/quick-start/3-configure-box-app.md]: If we're going to create a dev token for now, do we care if they have a JWT app? I guess we care cause we eventually want to add a guide on setting up your own downscope token server?

File: content/guides/mobile/ios/quick-start/3-configure-box-app.md [hunk 20]
```diff
@@ -0,0 +1,82 @@
...
+  </Choose>
+
+  <Choose option='ios.app_type' value='use_own' color='red'>
+    # Use an existing app
+
+    Use one of your existing Box JWT applications
+  </Choose>
+</Grid>
+
+<Choice option='ios.app_type' value='use_own' color='blue'>
```
Comment (cbetta) [inline content/guides/mobile/ios/quick-start/3-configure-box-app.md]: The value here needs to be `create_new`

File: content/guides/mobile/ios/quick-start/4-make-api-call.md [hunk 21]
```diff
@@ -0,0 +1,127 @@
...
+
+With our Box and iOS applications created and configured with the **Box iOS
+SDK** we can now make our first call to Box APIs.
+
+Using our blank iOS application, we will create a button to trigger
+a request to Box to fetch the name of the currently authenticated user.
+
+<Message warning>
+  As you integrate Box API calls into your iOS app, this blocking method
+  should be replaced with proper task delegation and non-blocking actions.
```
Comment (cbetta) [inline content/guides/mobile/ios/quick-start/4-make-api-call.md]: Not sure this read right. Maybe rephrase to actually talk about the implication? "Using a blocking actions as we use in these examples is slow. In a production app these blocking actions would need to be replaced with..."

File: content/guides/mobile/ios/quick-start/4-make-api-call.md [hunk 22]
```diff
@@ -0,0 +1,127 @@
...
+.padding(10)
+.cornerRadius(20)
+.foregroundColor(.white)
+.background(Color.blue)
+```
+
+Our next step is to add an an action for the button, which will be our API call
+to Box.
+
+## Add an API call button action
```
Comment (cbetta) [inline content/guides/mobile/ios/quick-start/4-make-api-call.md]: Replace with maybe: "Fetch a user's details"?

File: content/guides/mobile/ios/quick-start/4-make-api-call.md [hunk 23]
```diff
@@ -0,0 +1,127 @@
...
+
+We need to do two things add add the call to the Box API, add the import for
```
Comment (cbetta) [inline content/guides/mobile/ios/quick-start/4-make-api-call.md]: "When a user clicks the button, we want to fetch the user's details. To achieve this we need to do two things..."

File: content/guides/mobile/ios/quick-start/4-make-api-call.md [hunk 24]
```diff
@@ -0,0 +1,127 @@
...
+Next, within the button action, where we currently have a comment placeholder,
+add a call to the iOS SDK to fetch the current user. When the API call
+completes it will print an authentication message to the developer console. For
+ease of implementation, we have a blocking `sleep(5)` call in place in order to
+test that the iOS SDK can make calls from our iOS SDK by providing enough time
+for the request to complete.
+
+Replace `{{YOUR DEVELOPER TOKEN}}` with your developer token.
+
+```js
```
Comment (cbetta) [inline content/guides/mobile/ios/quick-start/4-make-api-call.md]: is this JS or swift?

File: content/guides/mobile/ios/quick-start/4-make-api-call.md [hunk 25]
```diff
@@ -0,0 +1,127 @@
...
+// Authenticate new Box Client
```
Comment (cbetta) [inline content/guides/mobile/ios/quick-start/4-make-api-call.md]: Please don't use inline comments. They can't be translated.

File: content/guides/mobile/ios/quick-start/4-make-api-call.md [hunk 26]
```diff
@@ -0,0 +1,127 @@
...
+let client = BoxSDK.getClient(token: "{{YOUR DEVELOPER TOKEN}}")
+
+// Get currently authenticated user
+client.users.getCurrent(fields:["name", "login"]) { (result: Result<User,
+  BoxSDKError>) in
+  guard case let .success(user) = result else {
+      print("Error getting user information")
+      return
+  }
+  print("Authenticated as \(user.name)")
```
Comment (cbetta) [inline content/guides/mobile/ios/quick-start/4-make-api-call.md]: Some english is fine.

File: content/guides/mobile/ios/quick-start/4-make-api-call.md [hunk 27]
```diff
@@ -0,0 +1,127 @@
...
+sleep(5)
+```
+
+Build and run your sample application in the iOS emulator. 
+
+<Message warning>
+  If you run this code an hour or more after you created the developer
+  token in the last step, you will need to revoke and generate a new developer
+  token using the same method in the
+  [last step](g://mobile/ios/quick-start/configure-box-app/) as the developer
```
Comment (cbetta) [inline content/guides/mobile/ios/quick-start/4-make-api-call.md]: *previous step

File: content/guides/mobile/ios/quick-start/4-make-api-call.md (hunk 24 above)
Comment (jcleblanc) [reply content/guides/mobile/ios/quick-start/4-make-api-call.md]: Swift, but didn't see that on the available markdown language list I was looking at. Will try with Swift.

Comment (cbetta) [reply content/guides/mobile/ios/quick-start/4-make-api-call.md]: Just use swift. I can set up a mapping to a recognised prism language.

File: content/guides/mobile/ios/quick-start/4-make-api-call.md (hunk 25 above)
Comment (jcleblanc) [reply content/guides/mobile/ios/quick-start/4-make-api-call.md]: Removed

File: content/guides/mobile/ios/quick-start/3-configure-box-app.md (hunk 19 above)
Comment (jcleblanc) [reply content/guides/mobile/ios/quick-start/3-configure-box-app.md]: Yeah, let's take an opinionated approach to this based on what we know they'll eventually need. Also removes the confusion of a choice.

File: content/guides/mobile/android/install.md (hunk 2 above)
Comment (jcleblanc) [reply content/guides/mobile/android/install.md]: Linked to the Github repo readme

File: content/guides/mobile/android/index.md (hunk 1 above)
Comment (jcleblanc) [reply content/guides/mobile/android/index.md]: Added image + additional text

File: content/guides/mobile/ios/index.md (hunk 4 above)
Comment (jcleblanc) [reply content/guides/mobile/ios/index.md]: Added image + additional text

File: content/guides/mobile/index.md (hunk 3 above)
Comment (jcleblanc) [reply content/guides/mobile/index.md]: Added some text but don't have a great image to use here.

File: content/guides/mobile/ios/quick-start/0-index.md (hunk 5 above)
Comment (jcleblanc) [reply content/guides/mobile/ios/quick-start/0-index.md]: Let me revisit this once I have a few more applications prepped to be able to show something nice.

File: content/guides/mobile/ios/quick-start/1-create-ios-app.md (hunk 6 above)
Comment (jcleblanc) [reply content/guides/mobile/ios/quick-start/1-create-ios-app.md]: Removed the second choice - now just a straightforward guide.

File: content/guides/mobile/ios/quick-start/1-create-ios-app.md (hunk 8 above)
Comment (jcleblanc) [reply content/guides/mobile/ios/quick-start/1-create-ios-app.md]: removed the border - should hopefully now flow nicer.

File: content/guides/mobile/ios/quick-start/1-create-ios-app.md (hunk 7 above)
Comment (jcleblanc) [reply content/guides/mobile/ios/quick-start/1-create-ios-app.md]: Removed

File: content/guides/mobile/ios/quick-start/1-create-ios-app.md (hunk 10 above)
Comment (jcleblanc) [reply content/guides/mobile/ios/quick-start/1-create-ios-app.md]: Changed to bold

File: content/guides/mobile/ios/quick-start/1-create-ios-app.md (hunk 16 above)
Comment (jcleblanc) [reply content/guides/mobile/ios/quick-start/1-create-ios-app.md]: Removed

File: content/guides/mobile/ios/quick-start/3-configure-box-app.md (hunk 18 above)
Comment (jcleblanc) [reply content/guides/mobile/ios/quick-start/3-configure-box-app.md]: I'm not sure that we want to go down that far - I wanted to try to avoid as much app configuration as possible as I felt it detracted from the purpose of the quickstart. Happy to revisit if you feel strongly otherwise though.

**Comment by jcleblanc** [comment]
@cbetta changes / corrections should now all be in place.

**Comment by cbetta** [comment]
Had to manually merge it through a new branch. See #90. For future ref can you please work on an actual branch instead of a fork @jcleblanc ?
